--- linux-v5.3-rc5/drivers/mmc/host/sunxi-mmc.c	2019-08-22 20:06:32.487082136 +0200
+++ linux-v5.3-rc5/drivers/mmc/host/sunxi-mmc.c	2019-08-22 20:08:20.415078693 +0200
@@ -1394,15 +1394,17 @@ static int sunxi_mmc_probe(struct platfo
 				  MMC_CAP_ERASE | MMC_CAP_SDIO_IRQ;
 
 	/*
-	 * Some H5 devices do not have signal traces precise enough to
-	 * use HS DDR mode for their eMMC chips.
+	 * Some H5 and H6 devices do not have signal traces precise
+	 * enough to use HS DDR mode for their eMMC chips.
 	 *
 	 * We still enable HS DDR modes for all the other controller
 	 * variants that support them.
 	 */
 	if ((host->cfg->clk_delays || host->use_new_timings) &&
 	    !of_device_is_compatible(pdev->dev.of_node,
-				     "allwinner,sun50i-h5-emmc"))
+				     "allwinner,sun50i-h5-emmc") &&
+	    !of_device_is_compatible(pdev->dev.of_node,
+		    		     "allwinner,sun50i-h6-emmc"))
 		mmc->caps      |= MMC_CAP_1_8V_DDR | MMC_CAP_3_3V_DDR;
 
 	ret = mmc_of_parse(mmc);
