--- arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi.org	2023-07-23 09:18:09.175085722 +0200
+++ arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2023-07-23 09:18:30.742336403 +0200
@@ -1225,13 +1225,13 @@
 
 			trips {
 				cpu_alert: cpu-alert {
-					temperature = <85000>;
+					temperature = <95000>;
 					hysteresis = <2000>;
 					type = "passive";
 				};
 
 				cpu-crit {
-					temperature = <100000>;
+					temperature = <105000>;
 					hysteresis = <0>;
 					type = "critical";
 				};
