--- arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts.org	2021-12-27 23:36:57.537904169 +0100
+++ arch/arm64/boot/dts/allwinner/sun50i-h6-eachlink-h6mini.dts	2021-12-27 23:38:09.044827289 +0100
@@ -80,6 +80,27 @@
 	cpu-supply = <&reg_vcccpu>;
 };
 
+&cpu_opp_table {
+	opp@1896000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1896000000>;
+
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+		status = "disabled";
+	};
+	opp@1992000000 {
+		clock-latency-ns = <244144>; /* 8 32k periods */
+		opp-hz = /bits/ 64 <1992000000>;
+
+		opp-microvolt-speed0 = <1160000 1160000 1200000>;
+		opp-microvolt-speed1 = <1100000 1100000 1200000>;
+		opp-microvolt-speed2 = <1100000 1100000 1200000>;
+		status = "disabled";
+	};
+};
+
 &ac200_pwm_clk {
 	status = "okay";
 };
