diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index 5818465abba9..da4f773c37c1 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -108,7 +108,7 @@ scm_version()
 	# If we are at the tagged commit, we ignore it because the
 	# version is well-defined. If none of the attempted tags exist
 	# or were usable, $count is still empty.
-	if [ -z "${count}" ] || [ "${count}" -gt 0 ]; then
+	if [ ! -z "${count}" ] || [ "${count}" -gt 0 ]; then
 
 		# If only the short version is requested, don't bother
 		# running further git commands
